### PR TITLE
fix: add credential to get private repo

### PR DESCRIPTION
**Pull request by bbbmj**

### CODE REVIEW

**Comment by bbbmj** [comment]
@ddysher @zjj2wry PTAL

File: prow/cmd/hook/main.go [hunk 1]
```diff
@@ -133,6 +134,8 @@ func main() {
 	if err != nil {
 		logger.WithError(err).Fatal("Error getting git client.")
 	}
+	// set github cerdential to clone repo.
+	gitClient.SetCredentials(*githubUsername, oauthSecret)
```
Comment (deyuanix) [inline prow/cmd/hook/main.go:138]: why do you need user name?  shouldn't token suffice?

Comment (zjj2wry) [reply prow/cmd/hook/main.go:138]: user name used for ```git clone```, when use `git clone` will require user input username and token

**Comment by zjj2wry** [comment]
@bbbmj set releasenote will remove do-not-merge/release-note-label-needed label
```releasenote
NONE
```

**Comment by bbbmj** [comment]
@ddysher PTAL

File: prow/cmd/hook/main.go (hunk 1 above)
Comment (deyuanix) [reply prow/cmd/hook/main.go:138]: I guess my question is why upstream doesn't need username?  is it because of private repo?

**Pull request merged**